app/models/game.rb

```diff
@@ -38,8 +37,55 @@ def populate_game!
     end
   end
 
+  def in_check?
+    kings = self.pieces.find_by(type: 'King')
+    
```

Comment:
if you use the loop to iterate over the kings, you will either way execute both methods..


the whole method could be only one line:

`black_in_check? || white_in_check?` 😉